test/unit-tests/expression/parse.test.js

```diff
@@ -1465,6 +1465,10 @@ describe('parse', function () {
       assert.strictEqual(parseAndEval('true & false'), 0)
       assert.strictEqual(parseAndEval('false & true'), 0)
       assert.strictEqual(parseAndEval('false & false'), 0)
+
+      assert.strictEqual(parseAndEval('0 & undefined'), 0)
+      assert.strictEqual(parseAndEval('false & undefined'), 0)
+      assert.throws(function () { parseAndEval('true & undefined') }, TypeError)
```

Comment:
Can you also add unit tests that validate defining variables, like `b =2 and c = 3`, to verify whether the variables are put in the original scope correctly? I think the `createSubScope(scope, args)` in `OperatorNode` should not be there, but I'm not entirely sure.

Reply:
@smith120bh can you still have a look into adding tests with defining variables? Besides that I think the PR is ready to be merged, it looks really neat 👌 .

Reply:
ping @smith120bh any chance of looking into some tests with defining variables? (If not I can have a look)

Reply:
Addressed via e4f8f78809f6f09046974279bf0154a041fb7d72